addons/spreadsheet/static/src/pivot/pivot_model.js

```diff
@@ -460,19 +483,56 @@ export class SpreadsheetPivotModel extends PivotModel {
 
     /**
      * @override
+     * Override _prepareData to build the mapping "readGroupDates" <-> "normalizedDates"
      */
-    _getGroupValues(group, groupBys) {
-        return groupBys.map((groupBy) => {
+    _prepareData(rootGroup, groupSubdivisions, config) {
+        super._prepareData(rootGroup, groupSubdivisions, config);
+        for (const groupSubdivision of groupSubdivisions) {
+            for (const subGroup of groupSubdivision.subGroups) {
+                this._buildReadGroupDateMapping(subGroup, groupSubdivision.rowGroupBy);
+                this._buildReadGroupDateMapping(subGroup, groupSubdivision.colGroupBy);
+            }
+        }
+    }
+
+    _buildReadGroupDateMapping(group, groupBys) {
+        for (const groupBy of groupBys) {
             const { field, aggregateOperator } = this.parseGroupField(groupBy);
             if (this._isDateField(field)) {
-                return pivotTimeAdapter(aggregateOperator).normalizeServerValue(
+                const normalized = pivotTimeAdapter(aggregateOperator).normalizeServerValue(
                     groupBy,
                     field,
                     group
                 );
+
+                if (!this._readGroupDateValuesMapping[groupBy]) {
+                    this._readGroupDateValuesMapping[groupBy] = {};
+                }
+                this._readGroupDateValuesMapping[groupBy][group[groupBy]] = normalized.toString();
             }
-            return this._sanitizeValue(group[groupBy]);
-        });
+        }
+    }
+
+    _getNormalizedDateValueFromReadGroupDate(groupBy, date) {
+        if (date === false) {
```

Comment:
false?  What are the expected values?

Reply:
the value is `false` when the field is not set :wink: